Change provision to onboard and add stage options

jayteemo opened the pull request:
IRIS-7817
IRIS-7822

jayteemo committed:
add --stage parameter

IRIS-7822

jayteemo committed:
change provision to onboard

The term "onboard" is now used to describe adding a device to an nRF cloud account.
The term "provision" is reserved for nRF Cloud's new Provisioning Service.
IRIS-7817

jayteemo committed:
rename nrf_cloud_provision.py to nrf_cloud_onboard.py

IRIS-7817

plskeggs commented on python/modem-firmware-1.3+/nrf_cloud_device_mgmt.py:
```diff
@@ -183,7 +183,7 @@ def url_encode(token):
 
 def parse_args():
-    parser = argparse.ArgumentParser(description="nRF Cloud Device Provisioning",
+    parser = argparse.ArgumentParser(description="nRF Cloud FOTA Update Job Management",
```
Maybe `Manage FOTA Update Jobs` would be better?

plskeggs commented on python/modem-firmware-1.3+/README.md:
````diff
@@ -67,245 +67,29 @@ If the `rtt` option is specified, communication will be performed using [SEGGER'
 To use RTT, the device must be running the [Modem Shell](https://github.com/nrfconnect/sdk-nrf/tree/main/samples/nrf9160/modem_shell) sample application [built with the RTT overlay](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/nrf9160/modem_shell/README.html#segger-rtt-support).
 This script will optionally flash the modem shell application on startup if a hex file path is provided with the `mosh_rtt_hex` option.
 
-```
````
I notice the other scripts show the help but you removed all of the help from this one.

I would like the help back.

I'm fine with the removal of the verbose examples from this.

jayteemo committed:
remove --help output and adjust heading size

Removed --help output since it can be easily viewed in the source code for each script.
Adjusted size of headings to improve readability.

glarsennordic reviewed (approved):
Sorry for the delay on approving this

I'll re-base my PR on top of yours when it merges
